src/v0/destinations/gainsight/util.js

```diff
@@ -95,22 +99,25 @@ const updateGroup = async (payload, Config) => {
       },
       { destType: 'gainsight', feature: 'transformation' },
     );
-  } catch (error) {
-    let errMessage = '';
-    let errorStatus = 500;
-    if (error.response && error.response.data) {
-      errMessage = error.response.data.errorDesc;
-      errorStatus = error.response.status;
-    }
-    throw new NetworkError(`failed to update group ${errMessage}`, errorStatus, {
-      [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(errorStatus),
-    });
-  }
 
-  if (!resp || !resp.data || resp.status !== 200) {
-    throw new RetryableError('failed to update group');
-  }
-  return resp.data.data.records[0].Gsid;
+    if (!isHttpStatusSuccess(processedResponseGs.status) && processedResponseGs >= 400 && processedResponseGs < 500) {
+      let errMessage = '';
+      let errorStatus = 500;
+      if (processedResponseGs.response) {
+        errMessage = processedResponseGs.response.errorDesc;
+        errorStatus = processedResponseGs.status;
+      }
+      throw new NetworkError(`failed to create group ${errMessage}`, errorStatus, {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(errorStatus),
+      }, processedResponseGs.response);
+    }
+  
+    if (processedResponseGs.status !== 200) {
+      throw new RetryableError('failed to create group',  processedResponseGs.status, {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(processedResponseGs.status),
+      }, processedResponseGs.response);
+    }
+    return processedResponseGs.response?.data?.records[0].Gsid;
```

Comment:
This logic of reading response from `handleHttpRequest` function is repeating. Can we commonalise them into a single function that handles gainsight API responses ?